packages/core/src/fhirpath/utils.ts

```diff
@@ -296,8 +299,25 @@ export function fhirPathEquivalent(x: TypedValue, y: TypedValue): TypedValue[] {
   if (isQuantity(xValue) && isQuantity(yValue)) {
     return booleanToTypedValue(isQuantityEquivalent(xValue, yValue));
   }
+
+  if (xType === 'Coding' && yType === 'Coding') {
+    if (typeof xValue !== 'object' || typeof yValue !== 'object') {
+      return booleanToTypedValue(false);
+    }
+    // "In addition, for Coding values, equivalence is defined based on the code and system elements only.
+    // The version, display, and userSelected elements are ignored for the purposes of determining Coding equivalence."
+    // Source: https://hl7.org/fhir/fhirpath.html#changes
+
+    // Basically checking if one of the `Coding` resources has a system defined. If so, then the two's system values must be compared
+    const systemsAreEquivalent =
```

Comment:
If both are undefined, wouldn't the equality check return `true` even without this ternary?

Reply:
You're right, one of the downsides to coding late at night is that you try to put unnecessary ternaries everywhere